src/ragas/metrics/_summarization_metric.py

```diff
@@ -0,0 +1,412 @@
+from __future__ import annotations
+
+import logging
+import typing as t
+from dataclasses import dataclass, field
+from typing import Dict
+
+import json
+import numpy as np
+from langchain.pydantic_v1 import BaseModel
+
+from ragas.llms.output_parser import RagasoutputParser, get_json_format_instructions
+from ragas.llms.prompt import Prompt, PromptValue
+from ragas.metrics.base import EvaluationMode, MetricWithLLM
+
+if t.TYPE_CHECKING:
+    from langchain.callbacks.base import Callbacks
+
+logger = logging.getLogger(__name__)
+
+class ExtractTopicsResponse(BaseModel):
+    topics: t.Dict[str, str]
+
+class LinkSummaryTopicsResponse(BaseModel):
+    summary_topics: t.Dict[str, str]
+
+class GenerateQuestionsResponse(BaseModel):
+    questions: t.List[str]
+
+class GenerateAnswersResponse(BaseModel):
+    answers: t.List[str]
+
+
+_output_instructions_question_generation = get_json_format_instructions(
+    pydantic_object=GenerateQuestionsResponse
+)
+_output_instructions_answer_generation = get_json_format_instructions(
+    pydantic_object=GenerateAnswersResponse
+)
+_output_instructions_topics_extraction = get_json_format_instructions(
+    pydantic_object=ExtractTopicsResponse
+)
+_outpyt_instructions_link_summary_topics = get_json_format_instructions(
+    pydantic_object=LinkSummaryTopicsResponse
+)
+
+_output_parser_question_generation = RagasoutputParser(pydantic_object=GenerateQuestionsResponse)
+_output_parser_answer_generation = RagasoutputParser(pydantic_object=GenerateAnswersResponse)
+_output_parser_topics_extraction = RagasoutputParser(pydantic_object=ExtractTopicsResponse)
+_output_parser_link_summary_topics = RagasoutputParser(pydantic_object=LinkSummaryTopicsResponse)
+
+EXTRACT_TOPICS_INSTRUCTION = """\
+Based on the given text, extract some broad opics, events, concepts or other important broader areas. Then for each of those extracted topic, associate chunks of text that are related to it and make topic-chunk pairs.
+
+** Important
+Return in JSON format with key as 'topics' and a dictionary of topic-chunk pairs as the value. Only return the JSON.
+"""
+TEXT_EXTRACTION_TOPICS = Prompt(
```

Comment:
* The idea of topic distribution is interesting but I have some concerns about it 
    * The assumption here is that each topic in the text has to be equally important, which would not be true for many texts
    * The workflow for calculating it is very expensive in terms of tokens. 
    * Paradigms for metrics should be very simple to understand, this concept seems little jargon for users coming from non-data science backgrounds.

Reply:
Agreed. Removed it for now. Would come up with a lighter version sometime later.